drivers/serial/uart_esp32.c

```diff
@@ -530,35 +466,14 @@ static const DRAM_ATTR struct uart_driver_api uart_esp32_api = {
 #endif  /* CONFIG_UART_INTERRUPT_DRIVEN */
 };
 
-#define GPIO0_NAME COND_CODE_1(DT_NODE_HAS_STATUS(DT_NODELABEL(gpio0), okay), \
-			     (DT_LABEL(DT_INST(0, espressif_esp32_gpio))), (NULL))
-#define GPIO1_NAME COND_CODE_1(DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay), \
-			     (DT_LABEL(DT_INST(1, espressif_esp32_gpio))), (NULL))
-
-#define DT_UART_ESP32_GPIO_NAME(idx, pin) ( \
-	DT_INST_PROP(idx, pin) < 32 ? GPIO0_NAME : GPIO1_NAME)
-
 #define ESP32_UART_INIT(idx)						       \
+										\
+PINCTRL_DT_INST_DEFINE(idx);							\
+										\
 static const DRAM_ATTR struct uart_esp32_config uart_esp32_cfg_port_##idx = {	       \
 	.uart_num = DT_INST_PROP(idx, peripheral),	\
```

Comment:
Remove `uart_num`, it is not necessary anymore.
Also, remove `peripheral` from devicetree binding so #43715 can be closed.

Reply:
@sylvioalves Removal was done here, I'll open another PR to address `hal_espressif`'s side.